collector/config/configgrpc/configgrpc.go

```diff
@@ -0,0 +1,564 @@
+	}
+
+	if gss.MaxConcurrentStreams > 0 {
+		opts = append(opts, grpc.MaxConcurrentStreams(gss.MaxConcurrentStreams))
+	}
+
+	if gss.ReadBufferSize > 0 {
+		opts = append(opts, grpc.ReadBufferSize(gss.ReadBufferSize))
+	}
+
+	if gss.WriteBufferSize > 0 {
+		opts = append(opts, grpc.WriteBufferSize(gss.WriteBufferSize))
+	}
+
+	// The default values referenced in the GRPC docs are set within the server, so this code doesn't need
+	// to apply them over zero/nil values before passing these as grpc.ServerOptions.
+	// The following shows the server code for applying default grpc.ServerOptions.
+	// https://github.com/grpc/grpc-go/blob/120728e1f775e40a2a764341939b78d666b08260/internal/transport/http2_server.go#L184-L200
+	if gss.Keepalive != nil {
+		if gss.Keepalive.ServerParameters != nil {
+			svrParams := gss.Keepalive.ServerParameters
+			opts = append(opts, grpc.KeepaliveParams(keepalive.ServerParameters{
+				MaxConnectionIdle:     svrParams.MaxConnectionIdle,
+				MaxConnectionAge:      svrParams.MaxConnectionAge,
+				MaxConnectionAgeGrace: svrParams.MaxConnectionAgeGrace,
+				Time:                  svrParams.Time,
+				Timeout:               svrParams.Timeout,
+			}))
+		}
+		// The default values referenced in the GRPC are set within the server, so this code doesn't need
+		// to apply them over zero/nil values before passing these as grpc.ServerOptions.
+		// The following shows the server code for applying default grpc.ServerOptions.
+		// https://github.com/grpc/grpc-go/blob/120728e1f775e40a2a764341939b78d666b08260/internal/transport/http2_server.go#L202-L205
+		if gss.Keepalive.EnforcementPolicy != nil {
+			enfPol := gss.Keepalive.EnforcementPolicy
+			opts = append(opts, grpc.KeepaliveEnforcementPolicy(keepalive.EnforcementPolicy{
+				MinTime:             enfPol.MinTime,
+				PermitWithoutStream: enfPol.PermitWithoutStream,
+			}))
+		}
+	}
+
+	var uInterceptors []grpc.UnaryServerInterceptor
+	var sInterceptors []grpc.StreamServerInterceptor
+
+	if gss.Auth != nil {
+		authenticator, err := gss.Auth.GetServerAuthenticator(host.GetExtensions())
+		if err != nil {
+			return nil, err
+		}
+
+		uInterceptors = append(uInterceptors, func(ctx context.Context, req any, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp any, err error) {
+			return authUnaryServerInterceptor(ctx, req, info, handler, authenticator)
+		})
+		sInterceptors = append(sInterceptors, func(srv any, ss grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+			return authStreamServerInterceptor(srv, ss, info, handler, authenticator)
+		})
+	}
+
+	if gss.MemoryLimiter != nil {
+		memoryLimiter, err := getMemoryLimiterExtension(gss.MemoryLimiter, host.GetExtensions())
+		if err != nil {
+			return nil, err
+		}
+
+		uInterceptors = append(uInterceptors, memoryLimiter.UnaryInterceptorGenerator())
+		sInterceptors = append(sInterceptors, memoryLimiter.StreamInterceptorGenerator()) 
```

Comment:
This is where the memorylimiter interceptors get added. Note that it is the responsibility of the extension to provide the StreamServerInterceptor and UnaryServerInterceptor functions.